clar2wasm/src/words/bindings.rs

```diff
@@ -29,6 +29,15 @@ impl ComplexWord for Let {
             let name = pair.get_name(0)?;
             let value = pair.get_expr(1)?;
 
+            // make sure name does not collide with builtin or user-defined symbols
+
+            if generator.is_reserved_name(name) {
+                return Err(GeneratorError::InternalError(format!(
```

Comment:
What you think in implementing a specific error type for naming clashes? `GeneratorError::BindingError`? 

Seems that this way it can be better reused through the codebase and also avoids hardcoded message strings.